doc/users/interactive.rst

```diff
@@ -146,45 +139,43 @@ Interactive mode controls:
 
 - whether created figures are automatically shown
 - whether changes to artists automatically trigger re-drawing existing figures
-- whether `.pyplot.show` returns immediately or after all of the
-  figures have been closed when given no arguments
-
+- when `.pyplot.show` exits: immediately, or after all of the figures have been closed if given no arguments
 
-If in interactive mode, then:
+Interactive mode:
 
-- newly created figures will be shown immediately
-- figures will automatically redraw on change
-- pyplot.show will return immediately by default
+- newly created figures will be displayed immediately
+- figures will automatically redraw when elements are changed
+- `pyplot.show` displays the figures and immediately returns to the prompt
 
-If not in interactive mode then:
+Not in interactive mode:
 
-- newly created figures and changes to figures will
-  not be reflected until explicitly asked to be
-- pyplot.show runs the GUI event loop and does not return until all of
-  the plot windows are closed
+- newly created figures and changes to figures are not displayed until 
+`.pyplot.show` is called again or `.pyplot.pause` exits
```

Comment:
The updates will only happen while `show` or `pause` are blocking and well cease after they return.  You can also get an update by calling `fig.canvas.flush_events` or by manually running the event loop.

Additionally, when not in interactive mode the stale hook is turned off so mutating the figure does not trigger `draw_idle`.

Reply:
I'm so unclear about what you're getting at here/how these suggestions translate into user facing docs.